rust/noosphere/src/ffi/noosphere.rs

```diff
@@ -72,3 +81,21 @@ pub fn ns_initialize(
 pub fn ns_free(noosphere: repr_c::Box<NsNoosphereContext>) {
     drop(noosphere)
 }
+
+#[ffi_export]
+/// De-allocate a Noosphere-allocated byte array
+pub fn ns_bytes_free(bytes: c_slice::Box<u8>) {
+    drop(bytes)
+}
+
+#[ffi_export]
+/// De-allocate a Noosphere-allocated string
+pub fn ns_string_free(string: char_p::Box) {
+    drop(string)
+}
+
+#[ffi_export]
+/// De-allocate a Noosphere-allocated array of strings
+pub fn ns_string_array_free(string_array: c_slice::Box<char_p::Box>) {
```

Comment:
This is a good idea, but I don't think we have a lot of flexibility when it comes to how aliased types are represented in the header. I'll play around with it.

Reply:
So, I tried this:

```rust
type NsString = char_p::Box;
type NsStringArray = c_slice::Box<NsString>;

#[ffi_export]
/// Read all header values for a file that correspond to a given name, returning
/// them as an array of strings
pub fn ns_sphere_file_header_values_read(
    sphere_file: &NsSphereFile,
    name: char_p::Ref<'_>,
) -> NsStringArray {
    // ...
}
```

This may be worth doing regardless, because it makes the Rust side of the FFI more clear. But, it doesn't change the generated C header at all, so the C side won't benefit from any documentation.

Reply:
Filed https://github.com/subconsciousnetwork/noosphere/issues/148